src/taipy/core/_scheduler/_scheduler_factory.py

```diff
@@ -14,27 +14,81 @@
 
+from ..common._utils import _load_fct
+from ..exceptions.exceptions import SchedulerNotBuilt
 from ._abstract_scheduler import _AbstractScheduler
+from ._dispatcher import _DevelopmentJobDispatcher, _JobDispatcher, _StandaloneJobDispatcher
 from ._scheduler import _Scheduler
-from ..common._utils import _load_fct
 
 
 class _SchedulerFactory:
 
     _TAIPY_ENTERPRISE_MODULE = "taipy.enterprise"
     _TAIPY_ENTERPRISE_CORE_MODULE = _TAIPY_ENTERPRISE_MODULE + ".core._scheduler._scheduler"
+    _dispatcher = None
+    _scheduler = None
 
     @classmethod
     def _build_scheduler(cls) -> Type[_AbstractScheduler]:
-        if Config.job_config.is_standalone or Config.job_config.is_development:
-            if util.find_spec(cls._TAIPY_ENTERPRISE_MODULE) is not None:
-                scheduler = _load_fct(
-                    cls._TAIPY_ENTERPRISE_CORE_MODULE,
-                    "Scheduler",
-                )
-            else:
-                scheduler = _Scheduler
+        # Build scheduler
+        if util.find_spec(cls._TAIPY_ENTERPRISE_MODULE) is not None:
+            cls._scheduler = _load_fct(
+                cls._TAIPY_ENTERPRISE_CORE_MODULE,
+                "Scheduler",
+            )
+        else:
+            cls._scheduler = _Scheduler
+
+        # Build dispatcher
+        if not cls._dispatcher:
+            cls._build_dispatcher()
+
+        # Initialize the scheduler
+        cls._scheduler.initialize()  # type: ignore
+
+        return cls._scheduler  # type: ignore
+
+    @classmethod
+    def _update_job_config(cls, force_restart=False):
+        cls._build_dispatcher(force_restart=force_restart)
+
+    @classmethod
+    def _build_dispatcher(cls, force_restart=False):
+        if not cls._scheduler:
+            raise SchedulerNotBuilt
+
+        if Config.job_config.is_standalone:
+            cls.__build_standalone_job_dispatcher(force_restart=force_restart)
+        elif Config.job_config.is_development:
+            cls.__build_development_job_dispatcher()
         else:
             raise ModeNotAvailable
-        scheduler.initialize()  # type: ignore
-        return scheduler  # type: ignore
+
+    @classmethod
+    def __build_standalone_job_dispatcher(cls, force_restart=False):
+        if isinstance(cls._dispatcher, _StandaloneJobDispatcher) and not isinstance(
+            cls._dispatcher, _DevelopmentJobDispatcher
+        ):
+            if force_restart:
+                cls._dispatcher.stop()
+                cls._dispatcher.join()  # wait for thread to be terminated TODO: investigate the effect
+            else:
+                return
+        cls._dispatcher = _StandaloneJobDispatcher(cls._scheduler)
+        cls._dispatcher.start()
+
+    @classmethod
+    def __build_development_job_dispatcher(cls):
+        if isinstance(cls._dispatcher, _StandaloneJobDispatcher) and not isinstance(
+            cls._dispatcher, _DevelopmentJobDispatcher
+        ):
```

Comment:
Another choice then Standalone or Development?

Reply:
Nope. At least not at the moment. There's also Airflow i think

Reply:
woo if there is Airflow, please remove it!  The Airflow Scheduler is not working anymore ^^
@jrobinAV

Reply:
Ah yes I believe it has been removed, I was just raising a point that there might be airflow in the future. As for now, we only have 2 :D